virttest/vt_netmgr/vt_netmgr.py

```diff
@@ -0,0 +1,194 @@
+
+        for obj in be_deleted:
+            self._networks.remove(obj)
+
+    def get_network_by_name(self, network_name, vm_name):
+        """
+        Filter the networks based on the network name and vm name.
+        NOTE: return all networks if descriptor is None.
+
+        :param network_name: The network name.
+        :type network_name: String.
+        :param vm_name: The vm name.
+        :type vm_name: String.
+
+        :return: The uuid of network or None If not found.
+        :rtype: String.
+        """
+        for obj in self._networks:
+            if network_name in (obj.name,) and vm_name in (obj.vm_name,):
+                return obj.uuid
+        return None
+
+    def query_network(self, uuid):
+        """
+        Get the network by an uuid.
+
+        :param uuid: The uuid.
+        :type uuid: String.
+
+        :return: The configuration of network or None if not found.
+                 e.g.: {"uuid": uuid,
+                        "name": name,
+                        "vm_name": vm_name,
+                        "type": type,
+                        "spec": spec}
+        :rtype: dict.
+        """
+        for obj in self._networks:
+            if uuid in (obj.uuid,):
+                return {
+                    "uuid": obj.uuid,
+                    "name": obj.name,
+                    "vm_name": obj.vm_name,
+                    "type": obj.type,
+                    "spec": obj.spec,
+                }
+        return None
+
+    def create_port(self, name, network_uuid, vm_name, mac_nic, net_type):
+        """
+        Create the port by its params.
+
+        :param name: The port name.
+        :type name: String.
+        :param network_uuid: The network uuid which this port belongs to.
+        :type network_uuid: String.
+        :param vm_name: The vm name.
+        :type vm_name: String.
+        :param mac_nic: The mac address.
+        :type mac_nic: String.
+        :param net_type: The net type.
+        :type net_type: String.
+        """
+        port_obj = port.Port(name, network_uuid, vm_name, mac_nic, net_type)
+        port_obj.create()
+        self._networks.append(port_obj)
+
+    def delete_port(self):
+        pass
+
+    # TODO:
+    # delete?
+    def get_port_by_name(self, nic_name, context):
+        """
+        Filter the ports based on the nic name and
+        context(vm name or network uuid).
+
+        :param nic_name: The network name or the uuid.
+        :type nic_name: String.
+        :param context: The vm name or network_uuid.
+        :type context: String.
+
+        :return: The uuid of network or None If not found.
+        :rtype: String.
+        """
+        for obj in self._ports:
+            if nic_name in (obj.name,) and (
+                context in (obj.vm_name,) or context in (obj.network_uuid,)
+            ):
+                return obj.uuid
+        return None
+
+    # TODO:
+    # delete?
+    def query_port(self, uuid):
```

Comment:
1. query_xxxx_info
2. merge the last 2 functions
3.  get_port_by_name => query_port